app/controllers/concerns/idv_session.rb

```diff
@@ -1,129 +1,29 @@
 module IdvSession
   extend ActiveSupport::Concern
 
-  if Figaro.env.proofing_vendors
-    Figaro.env.proofing_vendors.split(/\W+/).each do |vendor|
-      vendor_path = "#{Rails.root}/vendor/#{vendor}/lib"
-      $LOAD_PATH.unshift vendor_path
-    end
-  end
-
   def confirm_idv_session_started
-    redirect_to idv_session_url unless idv_session[:params].present?
+    redirect_to idv_session_url unless idv_session.params.present?
   end
 
   def confirm_idv_attempts_allowed
-    if idv_attempter.exceeded?
+    attempter = idv_session.attempter
+    if attempter.exceeded?
       flash[:error] = t('idv.errors.hardfail')
       redirect_to idv_fail_url
-    elsif idv_attempter.reset_attempts?
-      self.idv_attempts = 0
+    elsif attempter.reset_attempts?
+      idv_session.attempts = 0
     end
   end
 
-  def idv_session
-    user_session[:idv] ||= {}
-  end
-
-  protected
-
-  def idv_attempts=(num)
-    current_user.update!(idv_attempts: num)
-  end
-
-  def idv_attempts
-    current_user.idv_attempts
-  end
-
-  def idv_attempter
-    @_attempter ||= IdvAttempter.new(current_user)
-  end
-
-  def idv_flag_user_attempt
-    current_user.update!(idv_attempted_at: Time.zone.now)
-  end
-
-  def idv_question_number
-    idv_session[:question_number] ||= 0
-  end
-
-  def idv_resolution
-    idv_session[:resolution]
-  end
-
-  def idv_questions
-    idv_resolution.questions
-  end
-
-  def proofing_session_started?
-    idv_resolution.present? &&
-      idv_applicant.present? &&
-      idv_resolution.success?
-  end
-
-  def idv_vendor=(vendor)
-    idv_session[:vendor] = vendor
-  end
-
-  def idv_applicant=(applicant)
-    idv_session[:applicant] = applicant
+  def confirm_idv_needed
+    redirect_to idv_activated_url if current_user.active_profile.present?
```

Comment:
And having them separate means the new feature can land while you continue working on the refactoring.